src/Jellyfin.Database/Jellyfin.Database.Providers.SqLite/SqliteDatabaseProvider.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Jellyfin.Server.Implementations;
+using MediaBrowser.Common.Configuration;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Database.Providers.SqLite;
+
+/// <summary>
+/// Configures jellyfin to use an SQLite database.
+/// </summary>
+[JellyfinDatabaseProviderKey("Jellyfin-SQLite")]
+public sealed class SqliteDatabaseProvider : IJellyfinDatabaseProvider
+{
+    private readonly IApplicationPaths _applicationPaths;
+    private readonly ILogger<SqliteDatabaseProvider> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqliteDatabaseProvider"/> class.
+    /// </summary>
+    /// <param name="applicationPaths">Service to construct the fallback when the old data path configuration is used.</param>
+    /// <param name="logger">A logger.</param>
+    public SqliteDatabaseProvider(IApplicationPaths applicationPaths, ILogger<SqliteDatabaseProvider> logger)
+    {
+        _applicationPaths = applicationPaths;
+        _logger = logger;
+    }
+
+    /// <inheritdoc/>
+    public IDbContextFactory<JellyfinDbContext>? DbContextFactory { get; set; }
+
+    /// <inheritdoc/>
+    public void Initialise(DbContextOptionsBuilder options)
+    {
+        options.UseSqlite(
+            $"Filename={Path.Combine(_applicationPaths.DataPath, "jellyfin.db")};Pooling=false",
```

Comment:
Is pooling still a problem now?

Reply:
i had a reduced number of locks while disabling it.
Maybe thats really a candidate for an option in the settings later?